source/code/plugins/in_npmd_server.rb

```diff
@@ -182,15 +188,49 @@ def kill_all_agent_instances
                         # Process already stopped
                     rescue Errno::EPERM
                         # Trying to kill someone else's process?
-                        package_and_send_error_log("No perm to kill process with info:#{line}: our uid:#{Process.uid}")
+                        log_error "No perm to kill process with info:#{line}: our uid:#{Process.uid}"
                     rescue ArgumentError
                         # Could not get info on username
-                        package_and_send_error_log("Could not process username from info:#{line}: our uid:#{Process.uid}")
+                        log_error "Could not process username from info:#{line}: our uid:#{Process.uid}"
                     end
                 end
             end
         end
 
+        def diag_reader
+            begin
+                begin
+                    _line = @npmdDiagSock.gets
+                    if _line.nil? and !is_npmd_seen_in_ps()
+                        @npmdDiagSock = nil
+                        Logger::logInfo "Exiting diag thread as npmdAgent found stopped"
+                        break
+                    end
+                    next if _line.nil? or _line.strip== ""
+                    _json = check_and_get_json(_line.chomp)
+                    unless !_json.nil?
+                        Logger::logWarn "Sent string to diag is not a json string", Logger::loop
```

Comment:
Please try to not abbreviate the words , sometimes it makes the logs hard to read. Same comment applies to line#191.